sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -607,7 +607,12 @@ static int sof_audio_probe(struct platform_device *pdev)
 		hdmi_num = 3;
 	}
 
-	dmi_check_system(sof_rt5682_quirk_table);
+	if (dmi_check_system(sof_rt5682_quirk_table) == 0) {
+		/*dmi quirk has no matches */
+		if (pdev->id_entry) {
+			sof_rt5682_quirk = (unsigned long) pdev->id_entry->driver_data;
```

Comment:
I would flip the logic and use DMI last.

1. the quirk is set with a default in the machine driver
2. if present, you use driver_data
3. if a DMI quirk is present, use it.

That would allow us to still use DMI quirks even on TGL.

Reply:
Updated the patch set with this change, please review. Thanks!

Reply:
Updated the patchset, incorporated the suggestion.
Verified on both TGL, CML DUT's.